File: platform/College-1/Litcoder/Week 2/Code/lab6.java
import java.util.Scanner;

public class Main {
    private boolean isPrime(int num) {
        if (num == 0 || num == 1) {
            return true;
        }
        for (int i = 2; i < num; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Main ob = new Main();
        Scanner input = new Scanner(System.in);
        String str = input.next();
        int out = 0;
        int firstDigit = Character.getNumericValue(str.charAt(0));

        // Find the largest prime less than the first digit
        for (int iter = (firstDigit - 1); iter >= 1; iter--) {
            if (ob.isPrime(iter)) {
                out = out * 10 + iter;
                break;
            }
        }

        // Process the rest of the digits
        for (int count = 1; count < str.length(); count++) {
            for (int iter = 9; iter >= 1; iter--) {
                int temp = out * 10 + iter;
                if (ob.isPrime(temp)) {
                    out = temp;
                    break;
                }
            }
        }

        System.out.print(out);
    }
}

File: platform/College-1/Litcoder/Week 2/Code/lab6.py
def isPrime(num):
    if num == 0 or num == 1:
        return True
    for i in range(2, num):
        if num % i == 0:
            return False
    return True

num = input()
outNum = 0

# Find the largest prime less than the first digit
for i in range(int(num[0]) - 1, 0, -1):
    if isPrime(i):
        outNum = outNum * 10 + i
        break

# Process the rest of the digits
for _ in range(len(num) - 1):
    for digit in range(9, 0, -1):
        temp = outNum * 10 + digit
        if isPrime(temp):
            outNum = temp
            break

print(outNum, end='')
